### PR TITLE
Feat/msg data vaccination

### PR DESCRIPTION
**Pull request by mad-nuts**

### CODE REVIEW

File: iris-client-bff/src/test/java/iris/client_bff/iris_messages/web/IrisMessageControllerTest.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ class IrisMessageControllerTest {
 	private IrisMessageBuilderWeb irisMessageBuilder;
 
 	@MockBean
-	IrisMessageDataProcessor irisMessageDataProcessor;
+	EventMessageDataProcessor irisMessageDataProcessor;
```
Comment (jekutzsche) [inline iris-client-bff/src/test/java/iris/client_bff/iris_messages/web/IrisMessageControllerTest.java:58]: I find it awkward when a test of the general message implementation uses a class from a special implementation. Here in the concrete it is also prevented that the Event...Processor is limited in the visibility.

File: iris-client-bff/src/test/java/iris/client_bff/iris_messages/web/IrisMessageDataControllerTest.java [hunk 2]
```diff
@@ -47,7 +47,7 @@ class IrisMessageDataControllerTest {
 	private IrisMessageDataViewProvider irisMessageDataViewProvider;
 
 	@MockBean
-	IrisMessageDataProcessor irisMessageDataProcessor;
+	EventMessageDataProcessor irisMessageDataProcessor;
```
Comment (jekutzsche) [inline iris-client-bff/src/test/java/iris/client_bff/iris_messages/web/IrisMessageDataControllerTest.java:50]: As in `IrisMessageControllerTest.java`.

File: iris-client-bff/src/test/java/iris/client_bff/iris_messages/web/IrisMessageControllerTest.java (hunk 1 above)
Comment (mad-nuts) [reply iris-client-bff/src/test/java/iris/client_bff/iris_messages/web/IrisMessageControllerTest.java:58]: The generic message data tests are executed using event message data only. The IrisMessageDataProcessors class is mocked and can't be used to automatically resolve the correct Processor. When using the IrisMessageDataProcessor interface, the tests fail as there are 2 Processors now (Event & Vaccination). Is there any way to make it work by using the Interface?

**Review by jekutzsche — APPROVED** [review]

**Pull request merged**